src/fe/fe.C

```diff
@@ -283,6 +283,8 @@ void FE<Dim,T>::reinit(const Elem * elem,
         this->compute_shape_functions (elem,*pts);
       else
         this->compute_shape_functions(elem,this->qrule->get_points());
+      if (this->calculate_dual)
+        this->compute_dual_shape_functions();
```

Comment:
We we have something like a `libmesh_do_once`? @dewenyushu you asked about where you could put the warning...off the top of my head, I would put it here, something like:
```c++
{
  static bool warned = false;
  if (T != LAGRANGE && !warned)
  {
    libmesh_warning("dual calculations have only been verified for the LAGRANGE family");
    warned = true;
  }
  this->compute_dual_shape_functions();
}

Reply:
The `libmesh_warning()` macro is already wrapped in a `libmesh_do_once()` (see include/base/libmesh_common.h) so that outer static bool isn't necessary.

Reply:
Thanks! I just corrected it.